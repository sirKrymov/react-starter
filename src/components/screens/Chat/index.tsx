import React from 'react';

import { observer } from 'mobx-react-lite';

export const ChatScreen = observer(() => {
  return <div>Chat screen</div>;
});
