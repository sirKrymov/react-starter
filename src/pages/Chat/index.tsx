// @SETUP: Please don't forget to remove this component or current comment if component is used in project
import React, { ReactElement } from 'react';

import { ContentLimiter } from 'components/view/layouts/ContentLimiter';
import { ChatScreen } from 'components/screens/Chat';

export function ChatPage(): ReactElement {
  return (
    <ContentLimiter>
      <ChatScreen />
    </ContentLimiter>
  );
}
