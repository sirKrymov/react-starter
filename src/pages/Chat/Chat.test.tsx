import React from 'react';

import renderer, { ReactTestRenderer } from 'react-test-renderer';

import { ChatPage } from '../Chat';
import { BrowserRouter } from 'react-router-dom';

describe('<Chat> page component', () => {
  let component: ReactTestRenderer;

  const createComponent = () => {
    component = renderer.create(
      <BrowserRouter>
        <ChatPage />
      </BrowserRouter>
    );
  };

  describe('WHEN the component loads', () => {
    beforeEach(() => {
      createComponent();
    });

    it('THEN it should render without crashing', () => {
      expect(component.toJSON()).toBeDefined();
    });

    it('THEN it should match snapshot', () => {
      expect(component.toJSON()).toMatchSnapshot();
    });
  });
});
