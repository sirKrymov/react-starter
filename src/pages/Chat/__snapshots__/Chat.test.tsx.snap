// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Chat> page component WHEN the component loads THEN it should match snapshot 1`] = `
<div
  className="content-limiter"
>
  <div>
    Chat screen
  </div>
</div>
`;
